OpenSim/Simulation/Model/Geometry.h

```diff
@@ -0,0 +1,627 @@
+                        const SimTK::State& state) const;
+
+    /// Manage Appearance (how the Geometry is rendered) by applying Appearance 
+    /// from Geometry to DecorativeGeometry.
+    void setDecorativeGeometryAppearance(
+        SimTK::DecorativeGeometry& decoration) const {
+            decoration.setColor(get_Appearance().get_color());
+            decoration.setOpacity(get_Appearance().get_opacity());
+            decoration.setRepresentation(
+                (SimTK::DecorativeGeometry::Representation)
+                get_Appearance().get_representation());
+    };
+    /// Convenient access to set Appearance/Color
+    void setColor(const SimTK::Vec3& color) { 
+        upd_Appearance().set_color(color); 
+    };
+    /// Convenient access to get Appearance/Color
+    const SimTK::Vec3& getColor() const { 
+        return get_Appearance().get_color(); 
+    };
+
+    /// Convenient access to set Appearance/Opacity
+    void setOpacity(const double opacity) { 
+        upd_Appearance().set_opacity(opacity); 
+    };
+    /// Convenient access to get Appearance/Opacity
+    const double getOpacity() { 
+        return get_Appearance().get_opacity(); 
+    };
+
+    /// Convenient access to set Appearance/representation
+    void setRepresentation(const DisplayPreference& rep) { 
+        upd_Appearance().set_representation(rep); 
+    };
+    /// Convenient access to get Appearance/representation
+    const DisplayPreference& getRepresentation() { return 
+        (const DisplayPreference&)get_Appearance().get_representation(); 
+    };
+
+    /// Map this Geometry into a list of primitives aka SimTK::DecorativeGeometry 
+    /// and return it in the passed in Array.
+    virtual void createDecorativeGeometry(
+        SimTK::Array_<SimTK::DecorativeGeometry>&) const {};
+    /// Implement method from Component interface. Subclasses only need to 
+    /// implement createDecorativeGeometry to generate an Array of 
+    /// SimTK::DecorativeGeometry. From then on, setting Transforms & Appearance 
+    /// is handled by the base class Geometry to avoid duplication. 
+    void generateDecorations
+        (bool                                       fixed,
+        const ModelDisplayHints&                    hints,
+        const SimTK::State&                         state,
+        SimTK::Array_<SimTK::DecorativeGeometry>&   appendToThis) const override final;
+
+    /// Methods to support frame as a connection, implement Component interface
+    void constructConnectors();
+
+private:
+    
+    void setNull()
+    {
+        setAuthors("Ayman Habib");
+    }    //=====================================================================
+};  // END of class Geometry
+
+/**
+ * LineGeometry is a utility class used to abstract a line segment.
+ * It is used by muscle segments so that it's as small and useful as possible.
+ */
+class OSIMSIMULATION_API LineGeometry : public Geometry
```

Comment:
@aseth1 We have used this object as is for a while now to display muscles in the API visualizer. For lines that cross/connect multiple bodies, everything is drawn in ground Frame.